x-pack/legacy/plugins/actions/server/lib/license_state.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { i18n } from '@kbn/i18n';
+import { Observable, Subscription } from 'rxjs';
+import { ILicense, LICENSE_CHECK_STATE } from '../../../../../plugins/licensing/common/types';
+import { assertNever } from '../../../../../../src/core/utils';
+import { PLUGIN } from '../constants/plugin';
+
+export interface ActionsLicenseInformation {
+  showAppLink: boolean;
+  enableAppLink: boolean;
+  message: string;
+}
+
+export class LicenseState {
+  private licenseInformation: ActionsLicenseInformation = this.checkLicense(undefined);
+  private subscription: Subscription | null = null;
+
+  private updateInformation(license: ILicense | undefined) {
+    this.licenseInformation = this.checkLicense(license);
+  }
+
+  public start(license$: Observable<ILicense>) {
```

Comment:
I see we call start straight after construction, so, can we call `license$.subscribe` in the constructor and that allows us to change `private subscription: Subscription | null = null;` to `private subscription: Subscription;` which means we can avoid some of those null checks?